tools/gui/qtcodegen.py

```diff
@@ -0,0 +1,306 @@
+
+        cursor = self.textAusgabe.textCursor()
+        cursor.movePosition(QtGui.QTextCursor.End)
+        cursor.insertHtml(str(text))
+        cursor.insertText(str("\n"))
+        self.textAusgabe.setTextCursor(cursor)
+        self.textAusgabe.ensureCursorVisible()
+        
+        
+    def showInputDlg(self, dlgType):
+        
+        if dlgType == "GEBURTSDATUM":
+           while True:
+                try:
+                    text, ok = QtWidgets.QInputDialog.getText(self, 'Geburtsdatum', 'Bitte trage nachfolgend dein Geburtsdatum im Format DD.MM.YYYY ein.\n'
+                        'Beispiel: 02.03.1982\n')
+                    if ok:
+                        geburtsdatum = str(text)
+                        validate_datum(geburtsdatum)
+                        self.worker.signalUpdateData.emit("GEBURTSDATUM",geburtsdatum)
+                        break
+                    else:
+                        self.worker.signalStop.emit()
+                        break
+                except ValidationError as exc:
+                    QtWidgets.QMessageBox.critical(self, "Geburtsdatum ungültiges Format", "Das Datum entspricht nicht dem richtigen Format (DD.MM.YYYY).")
+                    
+        elif dlgType == "SMSCODE":
+            text, ok = QtWidgets.QInputDialog.getText(self, 'SMS Code', 
+                'Du erhältst gleich eine SMS mit einem Code zur Bestätigung deiner Telefonnummer\n'
+                'Bitte trage nachfolgend den SMS-Code ein.\n'
+                'Beispiel: 551-550\n')
+            if ok:
+                sms_pin = str(text).replace("-", "")
+                self.worker.signalUpdateData.emit("SMSCODE",sms_pin) 
+        elif dlgType == "SMSCODE_OK":
+            QtWidgets.QMessageBox.information(self, "Erfolgreich", "Code erfolgreich generiert. Du kannst jetzt mit der Terminsuche fortfahren.")
+            
+        self.worker.signalUpdateData.emit("DUMMY","")
+
+ 
+
+    def closeEvent(self, event):
+        """
+        Wird aufgerufen, wenn die Anwendung geschlossen wird
+        """
+
+        if self.thread.isRunning():
+            res = QtWidgets.QMessageBox.warning(self, "Suche beenden", "Suche wirklich beenden?\n",
+                                                (QMessageBox.StandardButton.Ok | QMessageBox.StandardButton.Cancel))
+
+            if res != QMessageBox.StandardButton.Ok:
+                event.ignore()
+                return
+                
+            event.accept()
+            
+        #exit
+        self.parent.enableCodeBtn.emit() # enable Code Btn again
+        
+        #stop worker
+        self.worker.stop()
+        self.worker.deleteLater()
+        
+        #stop thread
+        self.thread.quit()
+        self.thread.wait()
+        self.thread.terminate()
+       
+        
+        # Streams wieder korrigieren, damit kein Fehler kommt
+        sys.stdout = sys.__stdout__
+        sys.stderr = sys.__stderr__
+
+        self.deleteLater()
```

Comment:
wäre glaub nicht schlecht wenn man des event noch akzeptiert
```suggestion
        self.deleteLater()
        event.accept()
```